docs/docs/api/LayoutAnimations/KeyframeAnimations.md

```diff
@@ -0,0 +1,179 @@
+---
+id: keyframeAnimations
+title: Keyframe Animations
+sidebar_label: Keyframe Animations
+---
+
+The document explains how you can define complex animation using simple and popular animation definitions schema - Keyframes.
+
+## How to define Keyframe animation?
+
+### 1. Import Keyframe
+
+```js
+import { Keyframe } from 'react-native-reanimated';
+```
+
+### 2. Create Keyframe object, define initial and final state
+
+In Keyframe's constructor pass object with definitions of your animation. Object keys correspond to animation progress,
+so to 0 assign the style, you want for your object at the beginning of the animation and to 100 assign the style you want for your object to have at the end of the animation.
+
+```js
+import { Keyframe } from 'react-native-reanimated';
+
+const keyframe = new Keyframe({
+    0: {
+      transform: [ { rotate: '0deg' }],
+    },
+    100: {
+      transform: [ { rotate: '45deg' }],
+    },
+  }
+```
+
+### 3. Add middle points
+
+Between edge points, you can define middle points in which you want your object to have certain style properties.
+Remember that for every keyframe you should provide the same set of style properties!
+
+```js
+import { Keyframe } from 'react-native-reanimated';
+
+const keyframe = new Keyframe({
+    0: {
+      transform: [ { rotate: '0deg' }],
+    },
+    45: {
+        transform: [{ rotate: '100deg' }]
+    },
+    100: {
+      transform: [ { rotate: '45deg' }],
+    },
+  }
+```
+
+### 4. Customize transitions using an easing function
+
+```js
+import { Keyframe, Easing } from 'react-native-reanimated';
+
+const keyframe = new Keyframe({
+    0: {
+      transform: [ { rotate: '0deg' }],
+    },
+    45: {
+        transform: [{ rotate: '100deg' }],
+        easing: Easing.exp,
+    },
+    100: {
+      transform: [ { rotate: '45deg' }],
+    },
+  }
+```
+
+## How to use keyframe animations?
+
+Currently, you can define animations using keyframes only for entry and exit animations. 
+
+### 1. Choose Animated Component which entering or exiting you want to animate
+```js
+    // AnimatedComponent - component created by createAnimatedComponent or imported from Reanimated
+    // keyframe - Keyframe object
+    <AnimatedComponent exiting={keyframe} >
+```
+### 2. Customize the animation
+```js
+    <AnimatedComponent exiting={keyframe.duration(3000).delay(200)} >
+```
+### 3. Make sure that your animated component is under an AnimatedLayout. If it's not then add AnimatedLayout somewhere above the component.
+```js
+    <AnimatedLayout> // +
+        <View> sth </View>
```

Comment:
Please use `<Text />` if you want to show the rendering component with text. Adding text between any other component is invalid in RN and will error.